actors/evm/src/interpreter/instructions/lifecycle.rs

```diff
@@ -1,17 +1,156 @@
+        return Err(StatusCode::ActorError(ActorError::assertion_failed(
+            "inicode not in memory range".to_string(),
+        )));
+    };
+
+    // bump nonce and flush state TODO before send
+    let mut nonce = 0;
+    platform.rt.transaction(|state: &mut State, _rt| {
+        // this may be redundant if we are compiling with checked integer math
+        state.nonce = state.nonce.checked_add(1).unwrap();
+        nonce = state.nonce;
+        Ok(())
+    })?;
+
+    let params = CreateParams { code: input_data.to_vec(), nonce };
+
+    create_init(stack, platform, RawBytes::serialize(&params)?, CREATE_METHOD_NUM, value)
+}
+
+pub fn create2<'r, BS: Blockstore, RT: Runtime<BS>>(
+    state: &mut ExecutionState,
+    platform: &'r mut System<'r, BS, RT>,
 ) -> Result<(), StatusCode> {
-    todo!()
+    const CREATE2_METHOD_NUM: u64 = 3;
+    let ExecutionState { stack, memory, .. } = state;
+
+    // see `create()` overall TODOs
+
+    #[derive(Serialize_tuple, Deserialize_tuple)]
+    struct Create2Params {
+        #[serde(with = "strict_bytes")]
+        code: Vec<u8>,
+        salt: [u8; 32],
+    }
+
+    let endowment = stack.pop();
+    let (offset, size) = (stack.pop(), stack.pop());
+    let salt = stack.pop();
+
+    let endowment = TokenAmount::from(&endowment);
+    let input_region =
+        get_memory_region(memory, offset, size).map_err(|_| StatusCode::InvalidMemoryAccess)?;
+
+    // BE encoded array
+    let salt: [u8; 32] = salt.into();
+
+    let input_data = if let Some(MemoryRegion { offset, size }) = input_region {
+        &memory[offset..][..size.get()]
+    } else {
+        return Err(StatusCode::ActorError(ActorError::illegal_argument(
+            "initcode not in memory range".to_string(),
+        )));
+    };
+    let params = Create2Params { code: input_data.to_vec(), salt };
+
+    platform.rt.transaction(|state: &mut State, _rt| {
+        // this may be redundant if we are compiling with checked integer math
+        state.nonce = state.nonce.checked_add(1).unwrap();
+        Ok(())
+    })?;
+
+    create_init(stack, platform, RawBytes::serialize(&params)?, CREATE2_METHOD_NUM, endowment)
+}
+
+/// call into Ethereum Address Manager to make the new account
+fn create_init<'r, BS: Blockstore, RT: Runtime<BS>>(
+    stack: &mut Stack,
+    platform: &'r mut System<'r, BS, RT>,
+    params: RawBytes,
+    method: MethodNum,
+    value: TokenAmount,
+) -> Result<(), StatusCode> {
+    // send bytecode & params to EAM to generate the address and contract
+    let ret = platform.rt.send(&EAM_ACTOR_ADDR, method, params, value);
+
+    // https://github.com/ethereum/go-ethereum/blob/fb75f11e87420ec25ff72f7eeeb741fa8974e87e/core/vm/evm.go#L406-L496
+    // Normally EVM will do some checks here to ensure that a contract has the capability
+    // to create an actor, but here FVM does these checks for us, including:
+    // - execution depth (FVM) // TODO do we have different expectations here?
```

Comment:
Technically, but I think this is one of those cases where we'll just pick the FVM's values (same as gas).